### ISSUE TITLE
Enable ElasticSearch tests on Travis

### ISSUE THREAD

**Issue by ignasi35** [lagom/online-auction-java#48]
Use Travis CI's support for databases and ScalaTest tagging to run `search-impl` integration tests with ElasticSearch.

A possible issue will be the version miss-match between Travis options and our code/API client.

See also https://github.com/lagom/lagom/issues/569

**Comment by lakhina** [comment]
This issue involve setting up elasticsearch in travis.yml and adding one test to check if it is working. Right?

I will do it

**Comment by lakhina** [comment]
@ignasi35 Hey. Is this approach acceptable?
https://github.com/lakhina/online-auction-java/blob/master/.travis.yml#L5-L14
I liked it as we will not need to install any deb

**Comment by ignasi35** [comment]
How about you create a PR and we discuss it on the there? ;-)

Also, I've seen you are committing changes in your `master` branch, I recommend you create a branch per PR.

**Comment by lakhina** [comment]
@ignasi35 Yes sir, I just tried this commit on master because I dont know how to use travis on other branches. So I was testing my code with travis of my own master

**Comment by TimMoore** [comment]
Here are some ideas for writing tests that require Elasticsearch:

JUnit support [categories](https://github.com/junit-team/junit4/wiki/Categories). These are ways of tagging tests with a certain marker. We could use a category called `com.example.auction.search.test.ElasticsearchTests` and tag the test classes with that interface.

Then, as described in the [Using categories with SBT docs](https://github.com/junit-team/junit4/wiki/Categories#using-categories-with-sbt), we can provide `--exclude-categories=com.example.auction.search.test.ElasticsearchTests` to skip those tests.

The next step would be to configure that option by default, by modifying `build.sbt` as described in http://www.scala-sbt.org/0.13/docs/Testing.html#Test+Framework+Arguments We only need to provide that option in the `search-impl` project, so check the sbt docs for how to scope settings to a project. At this point, you can try running `sbt test` to be sure that the test doesn't run.

After that, we need the ability to optionally include the test. You can do this by overriding the sbt setting on the command line.

Something like this should do the trick:

```
sbt 'set testOptions in searchImpl in Test += Tests.Argument(TestFrameworks.JUnit, "--exclude-categories=")' test
```

You should confirm that running that command causes the search test to fail when Elasticsearch is _not_ running on your machine, and pass when it is.

Finally, modify `.travis.yml` to pass that command when running the tests, so that the search tests are enabled. Once you submit your pull request, you should be able to verify in the Travis CI output that the test is running (and passing).

For extra credit, you can use more advanced sbt configuration to make the command line a little nicer to read. You could define a custom setting key called `enableElasticsearch` that checks for a boolean system property with the same name and defaults to `false` if it's not set. Then, you could make the `testOptions` setting dependent on the `enableElasticsearch` setting. This requires a little bit more in depth sbt knowledge, and could be done in a later pull request.

**Comment by yg-apaza** [comment]
Whoa, this is so interesting! I achieved exclusion of Elasticsearch tests when running SBT's test task. Could I continue the work in another PR? Also, is there any way to check Travis CI output before doing the PR?

**Comment by ignasi35** [comment]
> Could I continue the work in another PR? 

Would you work together with @lakhina and share your work so you two get to the goal faster? 

> Also, is there any way to check Travis CI output before doing the PR? 

Anytime you push to a PR (even before merged) a Travis job is run and the output is available via the PR's list of checks passed/failed. Once a PR is merged, the list of checks is collapsed and harder to find but still available.
